pkg/handlers/index/alias_es.go

```diff
@@ -0,0 +1,184 @@
+
+outerLoop:
+	for _, action := range alias.Actions {
+		if action.Add != nil {
+			if action.Add.Index != "" {
+				matchAndAddToMap(indexList, action.Add.Index, addMap, action.Add)
+				continue outerLoop
+			}
+
+			// index is empty, try the indices field
+			for _, indexName := range action.Add.Indices {
+				matchAndAddToMap(indexList, indexName, addMap, action.Add)
+			}
+
+			continue outerLoop // this was an add action, don't bother checking action.Remove
+		}
+
+		if action.Remove != nil {
+			if action.Remove.Index != "" {
+				matchAndAddToMap(indexList, action.Remove.Index, removeMap, action.Remove)
+				continue outerLoop
+			}
+
+			// index is empty, try the indices field
+			for _, indexName := range action.Remove.Indices {
+				matchAndAddToMap(indexList, indexName, removeMap, action.Remove)
+			}
+		}
+	}
+
+	for alias, indexes := range addMap {
+		_ = core.ZINC_INDEX_ALIAS_LIST.AddIndexesToAlias(alias, indexes)
+	}
+
+	for alias, indexes := range removeMap {
+		_ = core.ZINC_INDEX_ALIAS_LIST.RemoveIndexesFromAlias(alias, indexes)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"acknowledged": true})
+}
+
+// @Id GetESAliases
+// @Summary Get index alias for compatible ES
+// @Tags    Index
+// @Produce json
+// @Param   index path  string  false  "Index"
+// @Param   alias path  string  false  "Alias"
+// @Success 200 {object} map[string]interface{}
+// @Failure 400 {object} meta.HTTPResponseError
+// @Router /es/{target}/_alias/{target_alias} [get]
+func GetESAliases(c *gin.Context) {
+	targetIndex := c.Param("target")
+
+	var targetIndexes []string
+	if targetIndex != "" {
+		targetIndexes = strings.Split(targetIndex, ",")
+	}
+
+	targetAlias := c.Param("target_alias")
+
+	var targetAliases []string
+	if targetAlias != "" {
+		targetAliases = strings.Split(targetAlias, ",")
+	}
+
+	m := core.ZINC_INDEX_ALIAS_LIST.GetAliasMap(targetIndexes, targetAliases)
+
+	c.JSON(http.StatusOK, m)
+}
+
+func indexNameMatches(name, indexName string) bool {
+	if name == indexName {
+		return true
+	}
+
+	if strings.Contains(name, "*") {
+		p, err := getRegex(name)
+		if err != nil {
+			log.Err(err).Msg("failed to compile regex")
+			return false
+		}
+
+		return p.MatchString(indexName)
+	}
+
+	return false
+}
+
+func getRegex(s string) (*regexp.Regexp, error) {
+	parts := strings.Split(s, "*")
+	pattern := ""
+	for i, part := range parts {
+		pattern += part
+		if i < len(parts)-1 {
+			pattern += "[a-zA-Z0-9_.-]+"
+		}
+	}
+
+	p, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
+	}
+
+	return p, nil
+}
+
+func matchAndAddToMap(indexList []*core.Index, indexName string, m map[string][]string, b *base) {
+	for _, index := range indexList {
+		n := index.GetName()
+		if indexNameMatches(indexName, n) {
+			if b.Alias != "" { // alias takes precedence over aliases
+				m[b.Alias] = append(m[b.Alias], n)
+			} else {
+				for _, a := range b.Aliases {
+					m[a] = append(m[a], n)
+				}
+			}
+		}
+	}
+}
```

Comment:
There is no need range indexList every time, just need range for if indexName is a wildcard.

Reply:
Thanks